### ISSUE TITLE
chore: Keyman 18.0 API changes

### ISSUE THREAD

**Issue by ermshiperete** [fcitx/fcitx5-keyman#7]
Keyman 18.0 has changed the way keyboards are loaded: instead of passing the name of a file, Keyman Core now expects to get a blob with the content of the file. This is in order to remove filesystem access from Core.

The existing API method `km_core_keyboard_load` has been/will be removed and replaced by `km_core_keyboard_load_from_blob`.

See https://github.com/keymanapp/keyman/issues/12497 and https://github.com/keymanapp/keyman/pull/12769.

**Comment by wengxt** [comment]
@ermshiperete to clarify a few things,
is there any file structure changes of keyboard files?
is it ok to use mmap? (I see your change uses fopen, any concerns on top of that?)

**Comment by ermshiperete** [comment]
@wengxt No, there aren't any changes to the file format. And I think it should be ok to use mmap.
